### PR TITLE
Add custom landing page for Help

### PR DESCRIPTION
**Pull request by azangru**
## Type
- New feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1155

## Description
- Updated the `Step` component to use the `label` field instead of `title`, and to accept children
- Added a custom Help landing page
- Hiding the rest of the pages of the Help app in production
- Allow users to see the help landing page in production (links on the Home page and in the launchbar)
- Fixed the styles on the About page

## Deployment URL
http://custom-help-landing.review.ensembl.org

## Views affected
- Help page
- About page

### CODE REVIEW

File: src/ensembl/src/content/app/help/Help.tsx [hunk 1]
```diff
@@ -31,28 +31,36 @@ import {
...
+    endpoint: `/api/docs/article?url=${encodeURIComponent(location.pathname)}`,
+    skip: isIndexPage
   });
+  const main = isIndexPage ? (
+    <main className={styles.main}>
+      <HelpLanding />
+    </main>
+  ) : article ? (
+    <MainContent article={article} />
+  ) : null;
```
Comment (msmanoj) [inline src/ensembl/src/content/app/help/Help.tsx:57]: Do we need the loading spinner here?

File: src/ensembl/src/content/app/help/Help.tsx [hunk 2]
```diff
@@ -63,26 +71,31 @@ const AppBar = () => {
...
+      )}
+    </HelpArticleGrid>
+  );
+
   return <main className={styles.main}>{content}</main>;
 };
 
+const isIndexRoute = (pathname: string) => {
+  // handle both /help and /help/
+  return pathname.split('/').filter(Boolean).join() === 'help';
```
Comment (msmanoj) [inline src/ensembl/src/content/app/help/Help.tsx:98]: Why not just `pathname.split('/')[1]`?

Comment (azangru) [reply src/ensembl/src/content/app/help/Help.tsx:98]: I think `pathname.replace('/', '')` might be even better

File: src/ensembl/src/content/app/help/Help.tsx (hunk 1 above)
Comment (azangru) [reply src/ensembl/src/content/app/help/Help.tsx:57]: The reason I added a loading spinner to the `About` component was that without it, the layout of the page was breaking while an article was loading: the content of the right column got shifted to the empty space to the left. With help pages, we don't have this problem — while an article is being loaded, the relevant part of the page just remains uniformly blank. I think we can add the spinner later if we want to.

File: src/ensembl/src/content/app/help/Help.tsx (hunk 2 above)
Comment (azangru) [reply src/ensembl/src/content/app/help/Help.tsx]: Updated.

File: src/ensembl/src/content/app/help/components/labelled-app-icon/LabelledAppIcon.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+type AppName = 'speciesSelector' | 'genomeBrowser' | 'entityViewer';
+
+type Size = 'regular' | 'small';
+
+type Props = {
+  app: AppName;
+  size: Size;
+};
+
+const appNameToComponent: Record<AppName, React.FunctionComponent> = {
```
Comment (msmanoj) [inline src/ensembl/src/content/app/help/components/labelled-app-icon/LabelledAppIcon.tsx:36]: would it be better to say `appNameToIconComponent`?

File: src/ensembl/src/header/launchbar/Launchbar.tsx [hunk 4]
```diff
@@ -74,28 +74,22 @@ const Launchbar = () => {
...
-              icon={CustomDownloadIcon}
-              enabled={
-                isEnvironment([
-                  Environment.DEVELOPMENT,
-                  Environment.INTERNAL
-                ]) && committedSpecies.length > 0
-              }
-            />
-          </div>
+          {isEnvironment([Environment.DEVELOPMENT, Environment.INTERNAL]) && (
```
Comment (msmanoj) [inline src/ensembl/src/header/launchbar/Launchbar.tsx:77]: Do we need to completely hide the custom download launchbar icon on production?

File: src/ensembl/src/content/app/help/Help.tsx (hunk 2 above)
Comment (msmanoj) [reply src/ensembl/src/content/app/help/Help.tsx]: I think we need `replaceAll` to handle both `/help` & `/help/`

File: src/ensembl/src/header/launchbar/Launchbar.tsx (hunk 4 above)
Comment (azangru) [reply src/ensembl/src/header/launchbar/Launchbar.tsx:77]: That was the idea, yes. We can verify with Andrea and Steve.

File: src/ensembl/src/content/app/help/Help.tsx (hunk 2 above)
Comment (azangru) [reply src/ensembl/src/content/app/help/Help.tsx]: 👍 ✅

File: src/ensembl/src/content/app/help/components/labelled-app-icon/LabelledAppIcon.tsx (hunk 3 above)
Comment (azangru) [reply src/ensembl/src/content/app/help/components/labelled-app-icon/LabelledAppIcon.tsx]: changed to `appNameToIcon`

**Review by msmanoj — APPROVED** [review]

**Review by imransl — APPROVED** [review]

**Pull request merged**